include/nbl/core/alloc/VectorViewNullMemoryResource.h

```diff
@@ -17,29 +17,34 @@ class VectorViewNullMemoryResource : public std::pmr::memory_resource
 {
     public:
         // only create the resource from an already sized vector
-        VectorViewNullMemoryResource(core::vector<uint8_t>&& buffer) : buffer(buffer)
+        VectorViewNullMemoryResource(core::vector<uint8_t>&& buffer) : buffer(std::move(buffer)), already_called(false)
         {
             assert(buffer.size());
-        };
+        }
 
         void* data() {
             return buffer.data();
         }
 
     protected:
         void* do_allocate(size_t bytes, size_t alignment) override {
-            assert(bytes <= buffer.size());
+            if (already_called || bytes > buffer.size() || !core::is_aligned_to(bytes, alignment))
```

Comment:
you're checking the wrong thing for alignment (its the `buffer.data()` you need to check!)